src/main/java/org/opensearch/security/privileges/DocumentWhitelist.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright OpenSearch Contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package org.opensearch.security.privileges;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.support.ConfigConstants;
+
+public class DocumentWhitelist {
```

Comment:
maybe a small javadoc header would help here which describes what this whitelist is being used for?

Reply:
I added a JavaDoc header